File: sample-bots/java/MyFirstLeader/MyFirstLeader.java
import dev.robocode.tankroyale.botapi.*;
import dev.robocode.tankroyale.botapi.events.*;
import dev.robocode.tankroyale.botapi.graphics.Color;

// ------------------------------------------------------------------
// MyFirstLeader
// ------------------------------------------------------------------
// A sample bot original made for Robocode by Mathew Nelson.
//
// Member of the MyFirstTeam. Looks around for enemies, and orders
// teammates to fire.
// ------------------------------------------------------------------
public class MyFirstLeader extends Bot {

    // The main method starts our bot
    public static void main(String[] args) {
        new MyFirstLeader().start();
    }

    // Called when a new round is started -> Leader's default behavior
    @Override
    public void run() {
        // Prepare robot colors to send to teammates
        var colors = new RobotColors();

        colors.bodyColor = Color.RED;
        colors.tracksColor = Color.CYAN;
        colors.turretColor = Color.RED;
        colors.gunColor = Color.YELLOW;
        colors.radarColor = Color.RED;
        colors.scanColor = Color.YELLOW;
        colors.bulletColor = Color.YELLOW;

        // Set the color of this robot containing the robot colors
        setBodyColor(colors.bodyColor);
        setTracksColor(colors.tracksColor);
        setTurretColor(colors.turretColor);
        setGunColor(colors.gunColor);
        setRadarColor(colors.radarColor);
        setScanColor(colors.scanColor);
        setBulletColor(colors.bulletColor);

        // Send RobotColors object to every member in the team
        broadcastTeamMessage(colors);

        // Set the radar to turn right forever
        setTurnRadarLeft(Double.POSITIVE_INFINITY);

        // Repeat while the bot is running
        while (isRunning()) {
            // Move forward and back
            forward(100);
            back(100);
        }
    }

    // Called when we scanned a bot -> Send enemy position to teammates
    @Override
    public void onScannedBot(ScannedBotEvent e) {
        // We scanned a teammate -> ignore
        if (isTeammate(e.getScannedBotId())) {
            return;
        }

        // Send enemy position to teammates
        broadcastTeamMessage(new Point(e.getX(), e.getY()));
    }


    // Called when we have been hit by a bullet -> turn perpendicular to the bullet direction
    @Override
    public void onHitByBullet(HitByBulletEvent e) {
        // Calculate the bullet bearing
        double bulletBearing = calcBearing(e.getBullet().getDirection());

        // Turn perpendicular to the bullet direction
        turnLeft(90 - bulletBearing);
    }
}

// ------------------------------------------------------------------
// Communication objects for team messages
// ------------------------------------------------------------------

// Point (x,y) class
class Point {
    public final double x;
    public final double y;

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }
}

// Robot colors
class RobotColors {
    public Color bodyColor;
    public Color tracksColor;
    public Color turretColor;
    public Color gunColor;
    public Color radarColor;
    public Color scanColor;
    public Color bulletColor;
}

File: sample-bots/python/MyFirstLeader/MyFirstLeader.py
from dataclasses import dataclass
from typing import Optional

from robocode_tank_royale.bot_api import Bot, team_message_type
from robocode_tank_royale.bot_api.color import Color
from robocode_tank_royale.bot_api.events import ScannedBotEvent, HitByBulletEvent


# ------------------------------------------------------------------
# Communication objects for team messages
# ------------------------------------------------------------------

# Point (x,y) class
@team_message_type
@dataclass
class Point:
    x: float
    y: float


# Robot colors
@team_message_type
@dataclass
class RobotColors:
    body_color: Optional[Color] = None
    tracks_color: Optional[Color] = None
    turret_color: Optional[Color] = None
    gun_color: Optional[Color] = None
    radar_color: Optional[Color] = None
    scan_color: Optional[Color] = None
    bullet_color: Optional[Color] = None


# ------------------------------------------------------------------
# MyFirstLeader
# ------------------------------------------------------------------
# A sample bot originally made for Robocode by Mathew Nelson.
#
# Member of the MyFirstTeam. Looks around for enemies, and orders
# teammates to fire.
# ------------------------------------------------------------------
class MyFirstLeader(Bot):

    def run(self) -> None:
        # Prepare robot colors to send to teammates
        colors = RobotColors()

        colors.body_color = Color.RED
        colors.tracks_color = Color.CYAN
        colors.turret_color = Color.RED
        colors.gun_color = Color.YELLOW
        colors.radar_color = Color.RED
        colors.scan_color = Color.YELLOW
        colors.bullet_color = Color.YELLOW

        # Set the color of this robot containing the robot colors
        self.body_color = colors.body_color
        self.tracks_color = colors.tracks_color
        self.turret_color = colors.turret_color
        self.gun_color = colors.gun_color
        self.radar_color = colors.radar_color
        self.scan_color = colors.scan_color
        self.bullet_color = colors.bullet_color

        # Send RobotColors object to every member in the team
        self.broadcast_team_message(colors)

        # Set the radar to turn left forever
        self.set_turn_radar_left(float("inf"))

        # Repeat while the bot is running: Move forward and back
        while self.running:
            self.forward(100)
            self.back(100)

    def on_scanned_bot(self, e: ScannedBotEvent) -> None:
        # We scanned a teammate -> ignore
        if self.is_teammate(e.scanned_bot_id):
            return
        # Send enemy position to teammates
        self.broadcast_team_message(Point(x=e.x, y=e.y))

    def on_hit_by_bullet(self, e: HitByBulletEvent) -> None:
        # Calculate the bullet bearing
        bullet_bearing = self.calc_bearing(e.bullet.direction)
        # Turn perpendicular to the bullet direction
        self.turn_left(90 - bullet_bearing)


def main() -> None:
    bot = MyFirstLeader()
    bot.start()


if __name__ == "__main__":
    main()
